src/main/java/io/vertx/core/http/impl/headers/HeadersMultiMap.java

```diff
@@ -337,6 +337,14 @@ public void forEach(Consumer<? super Map.Entry<String, String>> action) {
     }
   }
 
+  public void forEach(BiConsumer<String, String> action) {
```

Comment:
maybe we can have this in the `Multimap` API and instead pas `Handler<Map.Entry<CharSequence, CharSequence>>` ? (annotated with `@GenIgnore(GenIgnore.PERMITTED_TYPE)`)

Reply:
so this might lead to a class cast exception because sometimes entries can be not strings, e.g `io.netty.handler.codec.http.AsciiString`, instead we should have I think:

```java
  @Override
  public void forEach(Consumer<? super Map.Entry<String, String>> action) {
    HeadersMultiMap.MapEntry e = head.after;
    while (e != head) {
      Map.Entry<String, String> entry;
      if (e.key instanceof String && e.value instanceof String) {
        entry = (Map.Entry) e;
      } else {
        entry = new AbstractMap.SimpleEntry<>(e.key.toString(), e.value.toString());
      }
      action.accept(entry);
      e = e.after;
    }
  }
```

we should test this case which is the HTTP server case as headers are provided by Netty has `AsciiString` implementation of `CharSequence`

Reply:
Would that not have to be done in the existing `forEach` as well? 
I pretty much copied that and removed the allocation of the Map.Entry.

Reply:
perhaps we should have on MapEntry a method that transform it to a Map.Entry<String, String> so it can be called when we need it

Reply:
Added